### PR TITLE
mrc-330 Update spec

### PR DESCRIPTION
**Pull request by EmmaLRussell**
One outcome of the change to /report/.. rather than /reports/.. in the web endpoints is that there's now a discrepancy with the API endpoints, which are all still prefixed /reports/ - I'm assuming we're ok with that? I guess for Montagu it means that our api endpoints will look like https://whatever:port/reports/api/v1/reports/etc ..? Which is kind of weird but then API paths are often pretty horrible!

We also haven't listed the onetime_token endpoint. Do we still need that?

### CODE REVIEW

File: docs/spec/spec.md [hunk 1]
```diff
@@ -1,23 +1,20 @@
 # OrderlyWeb API
 
-Follows the general points in the [montagu api](https://github.com/vimc/montagu-api/blob/master/spec/spec.md)
+Follows the general points in the [montagu api](https://github.com/vimc/montagu-api/blob/master/docs/spec/spec.md)
```
Comment (EmmaLRussell) [inline docs/spec/spec.md]: This link was broken

File: docs/spec/spec.md [hunk 2]
```diff
@@ -1,23 +1,20 @@
...
+`https://[base-url]/api/v1/` e.g. `https://[base-url]/api/v1/reports/`
+* API endpoints can be accessed with or without trailing slashes e.g. both `https://[base-url]/api/v1/reports/` and 
+`https://[base-url]/api/v1/reports` will work. 
 
 In addition
 
 * Query parameters that accept booleans are case insensitive and accept `true` and `false`.
-* Authentication is via tokens issues by the Montagu API
-
-Some files are directly copied over (with only whitespace changes) from `montagu-api`:
```
Comment (EmmaLRussell) [inline docs/spec/spec.md]: This didn't really seem meaningful for a public-facing spec, for users who might not have had any experience of Montagu

File: docs/spec/spec.md [hunk 3]
```diff
@@ -59,13 +63,12 @@ Schema: [`LoginSuccessful.schema.json`](../schemas/LoginSuccessful.schema.json)
...
 
-Otherwise an error response is returned with status code 401 if the token was invalid, or 403 if the user is not
- a member of the configured GitHub organization or team.
+Otherwise an error response is returned with status code 401 if the token was invalid, or 403 if the authentication 
+provider is GitHub and the user is not a member of the configured GitHub organization or team.
 
 ## GET /reports/
 
-Return a list of all reports with minimal metadata - the id, human readable name, latest version of each and
-whether that version is published.
```
Comment (EmmaLRussell) [inline docs/spec/spec.md]: Pulished isn't part of the report but the version, and isn't returned by this endpoint - this is correct in the schema

File: docs/spec/spec.md (hunk 1 above)
Comment (hillalex) [reply docs/spec/spec.md]: Maybe we should remove this link and duplicate these "general points" in this spec, whatever they are! Does it just mean the below points or are there other points in the Montagu API spec that are missing here?

File: docs/spec/spec.md [hunk 4]
```diff
@@ -59,13 +64,20 @@ Schema: [`LoginSuccessful.schema.json`](../schemas/LoginSuccessful.schema.json)
...
     }
 
-Otherwise an error response is returned with status code 401 if the token was invalid, or 403 if the user is not
- a member of the configured GitHub organization or team.
+Otherwise an error response is returned with status code 401 if the token was invalid, or 403 if the authentication 
+provider is GitHub and the user is not a member of the configured GitHub organization or team.
+
+## GET /
+
+Return a description of all endpoint available over the api.
```
Comment (hillalex) [inline docs/spec/spec.md]: `endpoint` -> `endpoints`

File: docs/spec/spec.md [hunk 5]
```diff
@@ -327,16 +356,19 @@ Required permissions: `reports.read`.
 
 ## GET /reports/:name/versions/:version/all/
 
-Downloads a zip file of everything (including data).
+Downloads a zip file of all files associated with a report version (including data).
```
Comment (hillalex) [inline docs/spec/spec.md]: With `reports.read` this returns only artefacts, resources, and the readme (I think! double check code); with `reports.review` it's all files.

**Comment by hillalex** [comment]
Ooh, the onetime tokens. Hmm. I guess we don't need them anymore? Maybe we should open a ticket to remove them (pending investigation into whether they are still used/useful)

File: docs/spec/spec.md (hunk 4 above)
Comment (EmmaLRussell) [reply docs/spec/spec.md]: Good question re 'general points', I really wasn't sure what was meant by that! I'll have a look and see if I can glean anything...

File: docs/spec/spec.md (hunk 1 above)
Comment (EmmaLRussell) [reply docs/spec/spec.md]: Ah yeah, actually there is a list of general points, I'll copy that over

**Pull request merged**